tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestOutputHelperQueueTestOutputIntegration.cs

```diff
@@ -58,52 +60,78 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string
-
-                        return default;
-                    });
-            }
+                using var writer = LogFormatter.GetJsonWriter(sb);
 
-            private readonly struct Context
-            {
-                public readonly ulong TraceId;
-                public readonly ulong SpanId;
-                public readonly string Origin;
+                // Based on JsonFormatter
+                writer.WriteStartObject();
 
-                public Context(ulong traceId, ulong spanId, string origin)
+                writer.WritePropertyName("ddsource", escape: false);
+                writer.WriteValue("xunit");
+
+                if (HostMetadata.Instance.Hostname is { } hostname)
                 {
-                    TraceId = traceId;
-                    SpanId = spanId;
-                    Origin = origin;
+                    writer.WritePropertyName("hostname", escape: false);
+                    writer.WriteValue(hostname);
                 }
+
+                writer.WritePropertyName("timestamp", escape: false);
+                writer.WriteValue(DateTimeOffset.UtcNow.ToUnixTimeNanoseconds() / 1_000_000);
+
+                writer.WritePropertyName("message", escape: false);
+                writer.WriteValue(_message);
+
+                writer.WritePropertyName("status", escape: false);
+                writer.WriteValue("info");
+
+                string env = string.Empty;
+                if (_span is { } span)
+                {
+                    env = span.GetTag(Tags.Env) ?? string.Empty;
+
+                    writer.WritePropertyName("service", escape: false);
+                    writer.WriteValue(span.ServiceName);
+
+                    writer.WritePropertyName("dd.trace_id", escape: false);
+                    writer.WriteValue($"{span.TraceId}");
+
+                    writer.WritePropertyName("dd.span_id", escape: false);
+                    writer.WriteValue($"{span.SpanId}");
+
+                    if (span.GetTag(TestTags.Suite) is { } suite)
+                    {
+                        writer.WritePropertyName(TestTags.Suite, escape: false);
+                        writer.WriteValue(suite);
+                    }
+
+                    if (span.GetTag(TestTags.Name) is { } name)
+                    {
+                        writer.WritePropertyName(TestTags.Name, escape: false);
+                        writer.WriteValue(name);
+                    }
+
+                    if (span.GetTag(TestTags.Bundle) is { } bundle)
+                    {
+                        writer.WritePropertyName(TestTags.Bundle, escape: false);
+                        writer.WriteValue(bundle);
+                    }
+                }
+
+                // spaces are not allowed inside ddtags
+                env = env.Replace(" ", string.Empty);
+                writer.WritePropertyName("ddtags", escape: false);
+                writer.WriteValue($"env:{env},datadog.product:citest");
```

Comment:
What about other tags? Global tags are added automatically by `LogFormatter`, but will be ignored here?

Reply:
added in https://github.com/DataDog/dd-trace-dotnet/pull/3037/commits/ecf89fa2a08dec1f5a2215b0aadff5a8fef81f2c